draft-bbs-signatures.md

```diff
@@ -389,78 +387,102 @@ Procedure:
 This operation computes a deterministic signature from a secret key (SK) and optionally over a header and or a vector of messages.
 
 ```
-signature = Sign(SK, PK, header, (msg_1,..., msg_L), (H_1,..., H_L))
+signature = Sign(SK, PK, header, messages)
 
 Inputs:
 
-- SK (REQUIRED), an octet string of the form outputted by the KeyGen operation.
-- PK (REQUIRED), an octet string of the form outputted by the SkToPk operation provided the above SK as input.
-- header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
-- msg_1,..., msg_L (OPTIONAL), a vector of octet strings.
-- H_1,..., H_L (OPTIONAL), points of G1. Generators used to commit each message.
+- SK (REQUIRED), a non negative integer mod r outputted by the KeyGen
+                 operation.
+- PK (REQUIRED), an octet string of the form outputted by the SkToPk
+                 operation provided the above SK as input.
+- header (OPTIONAL), an octet string containing context and application
+                     specific information. If not supplied, it defaults
+                     to an empty string.
+- messages (OPTIONAL), a vector of octet strings.
```

Comment:
is this really optional?

Reply:
"Technically" yes. It is optional because someone could only sign the header (with always disclosed information) and not the messages (which you can selectively disclose from) and vice versa.

If they sign neither the result will be a signature on just the signers PK, which "technically" is valid (but most likely useless). To support the other 2 use cases though we left both as optional.